js/foam/apps/builder/controller/StackView.js

```diff
@@ -0,0 +1,577 @@
+          onRemove: function(ret, view, hints) {
+            if ( ! sanityCheck(view, hints, ['containerWidth']) ) return;
+            view.$.style.left = hints.containerWidth;
+            notifyCompleted(ret, view.$, 'left');
+          },
+          onResize: onResize.bind(this, true),
+        });
+        return {
+          slide: slideFromRight,
+          slideFromRight: slideFromRight,
+          slideFromLeft: this.Transition.create({
+            onAdd: function(ret, view, hints) {
+              if ( ! sanityCheck(view, hints, ['index', 'containerWidth']) ) return;
+              var style = view.$.style;
+              style.zIndex = hints.index + 1;
+              style.width = '0px';
+              style.left = '-' + hints.containerWidth + 'px';
+              ret();
+              style.transition = 'left 300ms ease';
+            },
+            onRemove: function(ret, view, hints) {
+              if ( ! sanityCheck(view, hints, ['containerWidth']) ) return;
+              view.$.style.left = '-' + hints.containerWidth + 'px';
+              notifyCompleted(ret, view.$, 'left');
+            },
+            onResize: onResize.bind(this, true),
+          }),
+          fade: this.Transition.create({
+            onAdd: function(ret, view, hints) {
+              if ( ! sanityCheck(view, hints, ['index']) ) return;
+              var style = view.$.style;
+              style.zIndex = hints.index + 1;
+              style.opacity = 0;
+              style.transition = 'opacity 300ms ease';
+              view.$.offsetLeft = view.$.offsetLeft;
+              style.opacity = 1;
+              notifyCompleted(ret, view.$, 'opacity');
+            },
+            onRemove: function(ret, view, hints) {
+              if ( ! sanityCheck(view) ) return;
+              view.$.style.opacity = 0;
+              notifyCompleted(ret, view.$, 'opacity');
+            },
+            onResize: onResize.bind(this, false),
+          }),
+          fadeOverlay: this.Transition.create({
+            onAdd: function(ret, view, hints) {
+              if ( ! sanityCheck(view, hints, ['index']) ) return;
+              var style = view.$.style;
+              style.zIndex = hints.index + 1;
+              style.opacity = 0;
+              style.transition = 'opacity 300ms ease';
+              view.$.offsetLeft = view.$.offsetLeft;
+              style.opacity = 1;
+              notifyCompleted(ret, view.$, 'opacity');
+            },
+            onRemove: function(ret, view, hints) {
+              if ( ! sanityCheck(view) ) return;
+              view.$.style.opacity = 0;
+              notifyCompleted(ret, view.$, 'opacity');
+            },
+          }),
+        };
+      },
+    },
+  ],
+
+  methods: [
+    function init(args) {
+      this.SUPER(args);
+      this.window.addEventListener('resize', this.onResize);
+    },
+    {
+      name: 'pushView',
+      documentation: function () {/* Default pushView that works on the top
```

Comment:
Methods docs can be specifiedinline: A leading /* ...*/ in the method body (before any code) is assumed to be documentation.